parity/main.rs

```diff
@@ -160,61 +168,120 @@ fn start() -> Result<String, String> {
 	}
 
 	let cmd = try!(conf.into_command());
-	execute(cmd)
+	execute(cmd, can_restart)
 }
 
+#[cfg(not(feature="stratum"))]
+fn stratum_main(_: &mut HashMap<String, fn()>) {}
+
 #[cfg(feature="stratum")]
-mod stratum_optional {
-	pub fn probably_run() -> bool {
-		// just redirect to the stratum::main()
-		if ::std::env::args().nth(1).map_or(false, |arg| arg == "stratum") {
-			super::stratum::main();
-			true
-		}
-		else { false }
-	}
+fn stratum_main(alt_mains: &mut HashMap<String, fn()>) {
+	alt_mains.insert("stratum".to_owned(), stratum::main);
 }
 
-#[cfg(not(feature="stratum"))]
-mod stratum_optional {
-	pub fn probably_run() -> bool {
-		false
-	}
+#[cfg(not(feature="ipc"))]
+fn sync_main(_: &mut HashMap<String, fn()>) {}
+
+#[cfg(feature="ipc")]
+fn sync_main(alt_mains: &mut HashMap<String, fn()>) {
+	alt_mains.insert("sync".to_owned(), sync::main);
 }
 
-#[cfg(not(feature="ipc"))]
-fn sync_main() -> bool {
-	false
+// TODO: merge with version in Updater.
+fn updates_path(name: &str) -> PathBuf {
+	let mut dest = PathBuf::from(env::home_dir().unwrap().to_str().expect("env filesystem paths really should be valid; qed"));
```

Comment:
Will be incompatible with @arkpar's directory structure changes. Setting `--db-path` should override placement into the user's home directory.